SU2_CFD/src/solver_direct_mean_inc.cpp

```diff
@@ -5309,56 +5240,77 @@ void CIncEulerSolver::BC_Sym_Plane(CGeometry      *geometry,
-    Area += Normal[iDim]*Normal[iDim];
-  Area = sqrt (Area);
-  
-  for (iDim = 0; iDim < nDim; iDim++)
-    UnitNormal[iDim] = -Normal[iDim]/Area;
-
-  /*--- Preprocessing: Compute unit tangential, the direction is arbitrary as long as t*n=0 ---*/
-  if (config->GetViscous()) {
-    switch( nDim ) {
-      case 2: {
-        Tangential[0] = -UnitNormal[1];
-        Tangential[1] =  UnitNormal[0];
-        break;
-      }
-      case 3: {
-        /*--- Find the largest entry index of the UnitNormal, and create Tangential vector based on that. ---*/
-        unsigned short Largest, Arbitrary, Zero;
-        if     (abs(UnitNormal[0]) >= abs(UnitNormal[1]) && 
-                abs(UnitNormal[0]) >= abs(UnitNormal[2])) {Largest=0;Arbitrary=1;Zero=2;}
-        else if(abs(UnitNormal[1]) >= abs(UnitNormal[0]) && 
-                abs(UnitNormal[1]) >= abs(UnitNormal[2])) {Largest=1;Arbitrary=0;Zero=2;}
-        else                                              {Largest=2;Arbitrary=1;Zero=0;}
-
-        Tangential[Largest] = -UnitNormal[Arbitrary]/sqrt(pow(UnitNormal[Largest],2) + pow(UnitNormal[Arbitrary],2));
-        Tangential[Arbitrary] =  UnitNormal[Largest]/sqrt(pow(UnitNormal[Largest],2) + pow(UnitNormal[Arbitrary],2));
-        Tangential[Zero] =  0.0;
-        break;
-      }
+  /*--- Get global marker value via the string Tag. ---*/
+  unsigned short iMarker_Global, 
+                 nMarker_Global = config->GetnMarker_CfgFile(),
+                 val_marker_Global;
+  for (iMarker_Global = 0; iMarker_Global < nMarker_Global; iMarker_Global++) {
+    if (config->GetMarker_All_TagBound(val_marker) == config->GetMarker_CfgFile_TagBound(iMarker_Global)) {
+      val_marker_Global = iMarker_Global; break;
     }
   }
   
   /*--- Loop over all the vertices on this boundary marker. ---*/
   for (iVertex = 0; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+    //necessary to acces the correct global val_marker here, currently it is wrong
+
+    if (iVertex == 0 || 
+        geometry->bound_is_straight[val_marker_Global] != true ||
+        config->GetKind_GridMovement() != RIGID_MOTION) {
+
+      /*---------------------------------------------------------------------------------------------*/
+      /*--- Preprocessing: On a symmetry-plane, the Unit-Normal is constant. Therefore a constant ---*/
+      /*---                Unit-Tangential to that Unit-Normal can be prescribed. The computation ---*/
+      /*---                of these vectors is done outside the loop (over all Marker-vertices).  ---*/
+      /*---                The "Normal" in SU2 is an Area-Normal and is most likely not constant  ---*/
+      /*---                on the symmetry-plane.                                                 ---*/
+      /*--- Edit July 2019: In order to use this BC_Sym_Plane method for slip walls in viscous    ---*/
+      /*---                 flow the unit-normal & tangent computation is moved into the loop over---*/
```

Comment:
Can you please remove the old comment in favor of the new? No need to keep the history, since git takes care of this for us.. might confuse folks more than it helps

Reply:
Done